lib/ansible/modules/database/mysql/mysql_db.py

```diff
@@ -62,6 +62,13 @@
     required: false
     default: []
     version_added: "2.7"
+  compact:
+    description:
+      - When I(state) is C(dump), generate compact dump. Equivalent to C(--compact) flag in C(mysqldump).
```

Comment:
Include a note about disabling compact resulting in the module always reporting changed.